Managers/UInputMan.cpp

```diff
@@ -939,172 +981,180 @@ namespace RTE {
 
 			// Enable the mouse cursor positioning again after having been disabled. Only do this when the mouse is within the drawing area so it
 			// won't cause the whole window to move if the user clicks the title bar and unintentionally drags it due to programmatic positioning.
-			int mousePosX = mouse_x / g_FrameMan.GetResMultiplier();
-			int mousePosY = mouse_y / g_FrameMan.GetResMultiplier();
-			if (m_DisableMouseMoving && m_PrepareToEnableMouseMoving && (mousePosX >= 0 && mousePosX < g_FrameMan.GetResX() && mousePosY >= 0 && mousePosY < g_FrameMan.GetResY())) {
+			int mousePosX = m_AbsoluteMousePos.m_X / g_WindowMan.GetResMultiplier();
+			int mousePosY = m_AbsoluteMousePos.m_Y / g_WindowMan.GetResMultiplier();
+			if (m_DisableMouseMoving && m_PrepareToEnableMouseMoving && (mousePosX >= 0 && mousePosX < g_WindowMan.GetResX() && mousePosY >= 0 && mousePosY < g_WindowMan.GetResY())) {
 				m_DisableMouseMoving = m_PrepareToEnableMouseMoving = false;
 			}
 		}
-		if (mousePlayer != Players::NoPlayer || !g_ActivityMan.IsInActivity()) {
-			// Mouse wheel update happens while a device is kb+mouse and while in the menus regardless of player devices.
-			// Translate motion into discrete ticks.
-			if (std::abs(mouse_z) >= 1) {
-				m_MouseWheelChange = mouse_z;
-				position_mouse_z(0);
-			} else {
-				m_MouseWheelChange = 0;
-			}
-		}
 	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	void UInputMan::UpdateJoystickInput() {
-		for (int joystick = 0; joystick < num_joysticks; ++joystick) {
-			// Detect and store joystick button changed since last Update()
-			for (int button = 0; button < joy[joystick].num_buttons; ++button) {
-				s_ChangedJoystickStates[joystick].button[button].b = joy[joystick].button[button].b != s_PrevJoystickStates[joystick].button[button].b;
+	void UInputMan::UpdateJoystickAxis(std::vector<Gamepad>::iterator device, int axis, int value) {
+		if (device != s_PrevJoystickStates.end()) {
+			int joystickIndex = device - s_PrevJoystickStates.begin();
+
+			int prevAxisValue = device->m_Axis[axis];
+			int prevDigitalValue = device->m_DigitalAxis[axis];
+
+			int newDigitalState = 0;
+			if (value > 8912) {
```

Comment:
What is 8912? Should probably be pulled into something named instead of being magic number